### ISSUE TITLE
Permission to Publish to Flathub

### ISSUE THREAD

**Issue by keplersj** [jorio/Bugdom#22]
Hi @jorio!

Thank you for porting this to modern operating systems and bundling an AppImage for Linux. I have created a [Flatpak](https://flatpak.org/) package for this project, ready to publish to [Flathub](https://flathub.org/home). With your permission I would like to publish the Flatpak I've created to Flathub to allow even broader access to this classic (including on the recently released Steam Deck).

Please let me know if there are any questions I can answer, thanks!

**Comment by jorio** [comment]
Hi @keplersj! Great initiative! As long as the build process can be audited, that's fine by me. It looks like everything happens in the open with Flatpak packages so it's all good.

Contrary to what BUILD.md currently says, I recommend using CMAKE_BUILD_TYPE "RelWithDebInfo" rather than "Release" because it might come in handy in case of crashes. I don't know Flatpak's policy regarding shipping debug symbols, though.

Also, for what it's worth, I have a series of bugfixes in the pipeline for Bugdom, and I intend to make a maintenance release probably next month. Do you plan to maintain the Flatpak package when updates come out?

**Comment by jorio** [comment]
Yes, please do the other Pangea games!

FYI, an update for Mighty Mike is right around the corner as well. It currently lacks a lot of the CI/packaging stuff that's in the other games, but you can use package name "io.jor.mightymike" for it.

I'm glad you're enjoying the games :) I also grew up with them and it's been lots of fun to get them up and running again!

By the way, now that you've mentioned the Steam Deck, I'm curious about how Bugdom fares on it. The menus and the pause screen currently require a mouse; does the Deck's touchscreen work well in those? If that's a problem, I can look into making the menus controller-friendly.

**Comment by keplersj** [comment]
Truthfully I'm not sure, I don't have a Deck just know Valve made Flatpak a first-class citizen in the immutable Arch it's preloaded with. If Deck touchscreen works anything like the touchscreen on the tablet (1st Gen Surface Go running Arch) I wrote the manifest on, the menus were decently touch friendly.

**Comment by keplersj** [comment]
Also, thought I'd share that the aarch64 build of Bugdom works pretty well on postmarketOS running Flatpak:
![20220315_15h49m12s_grim](https://user-images.githubusercontent.com/594124/158479114-0dfce814-2898-41bd-9631-196df2cd0ab2.png)
![20220315_15h49m56s_grim](https://user-images.githubusercontent.com/594124/158479116-bfb0e3f2-ed26-4e10-a34e-327e2b0f3741.png)
![20220315_15h50m12s_grim](https://user-images.githubusercontent.com/594124/158479121-a5596ffa-74fe-4856-923b-c477e9eeddc8.png)
![20220315_15h50m36s_grim](https://user-images.githubusercontent.com/594124/158479124-82440eee-b7cc-4d43-b31b-67b8cf10bead.png)

**Comment by jorio** [comment]
Cool! Thanks for sharing!

Can we try to get rid of the pixelation? If you have time, I wonder if adding `SDL_WINDOW_ALLOW_HIGHDPI` [to the flags passed to SDL_CreateWindow in Main.cpp](https://github.com/jorio/Bugdom/blob/master/src/Main.cpp#L153) makes a difference.

**Comment by keplersj** [comment]
Of course! Here's screenshots from a build with the following patch (https://github.com/keplersj/Bugdom/commit/070381a61fd0c3e5c9de6811b941821cc46810d9):
![20220315_16h52m47s_grim](https://user-images.githubusercontent.com/594124/158485549-e31eba7e-fbbf-4199-9027-12103ad3912a.png)
![20220315_16h53m00s_grim](https://user-images.githubusercontent.com/594124/158485557-5f4cbb54-9cbf-42eb-90dd-3de452c1e907.png)

**Comment by keplersj** [comment]
Of course! Here are screenshots from a build with the following patch applied (https://github.com/jorio/Bugdom/pull/23/commits/070381a61fd0c3e5c9de6811b941821cc46810d9):
![20220315_17h03m04s_grim](https://user-images.githubusercontent.com/594124/158486482-b5e90830-a64b-407e-8e3e-da79c20b9179.png)
![20220315_17h03m19s_grim](https://user-images.githubusercontent.com/594124/158486485-8cb5f8d9-74d2-4d50-85b3-b6bc078e9ac4.png)

**Comment by juxuanu** [comment]
@keplersj Thanks for the Flatpak. Any indications of how can I port my saved games there? I used to compile it myself with the PKGBUILD in the AUR, but the Flatpak is much more convenient. I allowed access to ~/.config/Bugdom but it did not detect them.

Thanks for your time and work!

**Comment by keplersj** [comment]
@juxuanu Very glad the Flatpak is working well for you! Moving the contents of `~/.config/Bugdom` to `~/.var/app/io.jor.bugdom/config/Bugdom` should port your previous scores, saves, and preferences.

**Comment by jorio** [comment]
@keplersj Do you have time to take a look at the Bugdom and Mighty Mike flatpaks? I've been getting notifications about them since I released updates for these games.

**Comment by jm2** [comment]
@keplersj in addition to the updates could you also please add Cro-Mag Rally to the set?

**Comment by Tagkit** [comment]
@keplersj & @jorio Idk if anyone has come forward to test the compatibility of this and the other games on the Steam Deck, but - while I'm still waiting on shipping for mine- I'd be willing to do some tests and suggestions for you both on the compatibility for these games once my Deck comes in. Anything in particular you are looking for / need feedback on?

**Comment by jm2** [comment]
What I've noticed in very short testing is that SteamInput doesn't appear to get passed to the Pangea apps installed through Flathub and launched via Steam on the Deck. The AppImage binaries provided here work without issue.

**Comment by jorio** [comment]
Let's discuss the Steam Deck in a new issue! (#29)
